rails/app/controllers/admin/themes_controller.rb

```diff
@@ -0,0 +1,27 @@
+require 'pry'
+module Admin
+  class ThemesController < Admin::ApplicationController
+    def create
+      if theme_params[:active]
```

Comment:
So, this check and the one on line 12 will *always* run the `deactivate_prev_theme` code even if active was not checked (is set to false) in the form.

This is because the form input is sending back "0" / "1" for false / true if the checkbox was checked or not, and so an object check will always return true because it's a string.

The quickest way to resolve this is to check if `theme_params[:active] == "1"`, which is the equivalent to true.